### PR TITLE
Feature/passm 17 create docker compose file

### PR DESCRIPTION
**Pull request by Atharva0418**
This pull request adds a Docker Compose file to simplify the setup and management of the Spring Boot application and its MySQL database. The configuration includes:

App Service:

- Uses the image atharva0418/springboot-gradle-app.
- Exposes port 3000 and sets environment variables for database connection.
- Depends on the MySQL service.

DB Service:

- Uses the official MySQL image.
- Creates a database named test with root password root.
- Exposes port 3307 on the host.
- Networking and Volumes:

- Configures a custom network (app-network) for inter-container communication.
- Uses a volume (mysql_data) to persist MySQL data.

Benefits: This configuration allows for easy deployment and consistent environment setup for all developers using a single command (docker-compose up).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+services:
+  app:
+    image: atharva0418/springboot-gradle-app
```
Comment (chintal31) [inline docker-compose.yml]: how is this image built? 
when I do docker compose up, it throws the error
"pull access denied for atharva0418/springboot-gradle-app, repository does not exist or may require 'docker login': denied: requested access to the resource is denied"

Can you list down steps on how you are running it locally? At the end, the goal is to run the app using `docker compose up` command.

Comment (Atharva0418) [reply docker-compose.yml]: It is built by the springboot feature 'bootBuildImage'. Check the build.gradle file and replace 'atharva0418' to your dockerhub username. Spring takes care of all the dependencies and configurations and builds an image of the application. Run `gradle bootBuildImage` command to build the image,if you have mentioned your username in the imagename it should be stored in your docker registry. After that you should be able to run the app using `docker compuse up` command.

Comment (chintal31) [reply docker-compose.yml]: I see, okay. Wasn't aware of this spring boot feature `bootBuildImage`. Great 👍 

I just renamed the imageName and the container name making it more generalized and one does not have to use dockerhub username. 

You can update the readme and add the section on how to run the app via docker.

**Review by keya5406 — APPROVED** [review]

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -10,9 +10,10 @@ spring.profiles.active=dev
 spring.datasource.url=${DB_URL}
 spring.datasource.username=${DB_USERNAME}
 spring.datasource.password=${DB_PASSWORD}
-spring.datasource.driver-class-name=${DB_DRIVER_CLASS_NAME}
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 
 spring.jpa.hibernate.ddl-auto=update
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQLDialect
```
Comment (chintal31) [inline src/main/resources/application.properties:16]: what is this used for?

**Review by chintal31 — APPROVED** [review]
Overall well done 👍 

LGTM

**Comment by chintal31** [comment]
Hi @keya5406 
It's okay if you don't get everything in this PR's. The goal is to get slowly slowly familiar with the terminology. For example, from this PR, you could check 

- what is Docker? 
- Docker Compose? 
- Docker Image? 
- Container?

@Atharva0418 you could also discuss the concepts with her that will also improve your understanding.

File: src/main/resources/application.properties (hunk 2 above)
Comment (Atharva0418) [reply src/main/resources/application.properties:16]: It tells hibernate that we are using MYSQL syntax so it does not confuse it with PostgreSQL or Oracle. If not defined, hibernate tries to identify it based on the database used, however it was throwing an error when this was not specified. So I specified it to get rid of the error.

**Pull request merged**